pymeasure/instruments/oxfordinstruments/ips120_10.py

```diff
@@ -0,0 +1,457 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+
+import logging
+from time import sleep, time
+import numpy
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, \
+    truncated_range, strict_range
+
+
+# Setup logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class MagnetError(ValueError):
+    """ Exception that is raised for issues regarding the state of the magnet or power supply. """
+    pass
+
+
+class SwitchHeaterError(ValueError):
+    """ Exception that is raised for issues regarding the state of the superconducting switch. """
+    pass
+
+
+class IPS120_10(Instrument):
+    """Represents the Oxford Superconducting Magnet Power Supply IPS 120-10.
+
+    .. code-block:: python
+
+        ips = IPS120_10("GPIB::25")  # Default channel for the IPS
+
+        ips.enable_control()         # Enables the power supply and remote control
+
+        ips.train_magnet([           # Train the magnet after it has been cooled-down
+            (11.8, 1.0),
+            (13.9, 0.4),
+            (14.9, 0.2),
+            (16.0, 0.1),
+        ])
+
+        ips.set_field(12)           # Bring the magnet to 12 T. The switch heater will
+                                    # be turned off when the field is reached and the
+                                    # current is ramped back to 0 (i.e. persistent mode).
+
+        print(self.field)           # Print the current field (whether in persistent or
+                                    # non-persistent mode)
+
+        ips.set_field(0)            # Bring the magnet to 0 T. The persistent mode will be
+                                    # turned off first (i.e. current back to set-point and
+                                    # switch-heater on); afterwards the switch-heater will
+                                    # again be turned off.
+
+        ips.disable_control()       # Disables the control of the supply, turns off the
+                                    # switch-heater and clamps the output.
+
+    :param clear_buffer: A boolean property that controls whether the instrument
+        buffer is clear upon initialisation.
+    :param switch_heater_delay: The time in seconds (default is 20s) to wait after
+        the switch-heater is turned on or off before the heater is expected to be
+        in the correct mode.
+    :param field_range: A numeric value or a tuple of two values to indicate the
+        lowest and highest allowed magnetic fields. If a numeric value is provided
+        the range is expected to be from :code:`-field_range` to :code:`+field_range`.
+
+    """
+
+    _MAX_CURRENT = 120  # Ampere
+    _MAX_VOLTAGE = 10  # Volts
+    _SWITCH_HEATER_DELAY = 20  # Seconds
+    _FIELD_RANGE = [-16, 16]  # Tesla
```

Comment:
I think it might be beneficial to reduce the default field range to a field range which is most common in magnets, and not necessarily one which is usually found in more special facilities. 
Therefore, I would recommend to change the range to [-7, 7]  # Tesla, since I don't know any magnets one would want to control with such a device which cannot achieve this.

Reply:
Fair enough, +-7 Tesla sounds like a decent range for most magnets.